aoc/d3/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+// split a string in two using the middle point
+func SplitString(input string) []string {
+	half := len(input) / 2
+	return []string{input[:half], input[half:]}
+}
+
+// find the character that exists in two strings
+func FindCommonItem(input1 string, input2 string) string {
+	var result string
+	for _, item := range input1 { // generated by ChatGPT
+		if i := strings.Index(input2, string(item)); i != -1 {
+			result = string(item)
+		}
+	}
+	return result
+}
```

Comment:
Couple things on this func. You can specify consecutive params of the same type with one type annotation after the last param.
Also, you can return as soon as you find a common item as there should only be one (anyway even if there was more than one the original function continues and returns only the last one).
Probably a good idea to return an error if there are no common items. And no need to assign the value of the index function if it's only going to be used in the assertion.
Finally, using `IndexRune` instead of `Index` avoids having to cast the rune into a string (I wanted to change the return value of this function to `rune` and avoid all casting, but then you'd have to rewrite other funcs too or cast it anyway afterwards. But technically speaking this function returns a rune and not a string.
```suggestion
func FindCommonItem(left, right string) (string, error) {
	for _, rune := range left {
		if strings.IndexRune(right, rune) != -1 {
			return string(rune), nil
		}
	}
	return "", errors.New("unable to find common item")
}
```

Reply:
edit: not casting, converting. Casting is different :)

Reply:
I like this a lot, thanks for the detailed explanation. The linter suggested used strings.ContainsRune instead of IndexRune and it works just fine. Added a test for exception handling too.
https://github.com/lucasrijllart/advent-of-code-2022/pull/1/commits/24ff2ac78c703702439b799d9687c1be9151342a